src/reply_options.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{properties::Property, types::BinaryData};
+
+/// Used to specify information about replies to received MQTT messages.
+pub struct ReplyOptions<'a> {
```

Comment:
Do we really need a struct for this? Can't this be just the original (or a new) slice of `&[Property]`?
That way you could pass the packet (or just the Property array/slice) to `reply()`.

Reply:
I had it originally implemented like this, but found the desire to:
1. Add support for an optional-default topic if no `ResponseTopic` was found (e.g. default logging API)
2. Add an "ignore-message-if-no-`ResponseTopic`" functionality

Those would both just add arguments into the `reply()` API until it became unwieldy, so I opted to wrap them up in `ReplyOptions` instead